packages/app/src/config.ts

```diff
@@ -1,7 +1,19 @@
-import type { ConfigurationOptions } from 'aws-sdk';
+import type { CloudWatchLogsClientConfig } from '@aws-sdk/client-cloudwatch-logs';
+import type { ECSClientConfig } from '@aws-sdk/client-ecs';
+import type { KinesisClientConfig } from '@aws-sdk/client-kinesis';
+import type { LambdaClientConfig } from '@aws-sdk/client-lambda';
+import type { S3ClientConfig } from '@aws-sdk/client-s3';
+
+type CommonAWSConfig = Pick<
+	S3ClientConfig,
+	keyof KinesisClientConfig &
+		keyof ECSClientConfig &
+		keyof CloudWatchLogsClientConfig &
+		keyof LambdaClientConfig
+>;
```

Comment:
I couldn't find an exported type that covered the common properties we use in this project, so I've tried to derive one here (although I don't think it's perfect, what if the configs share the same key but the value is a different type?).

I initially just hardcoded the properties we used, as a simpler solution:
```typescript
type CommonAWSConfig = {
	region?: string;
	maxRetries?: number;
};

```
BUT, this revealed a subtle API change that TypeScript wouldn't have caught - that `numRetries` changed to `maxAttempts`. [This playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBKYDMCWBzGBeGBvAUDfMATgKYpLgBcMARMQK4C0A7sdAwIzUA0eBYdAWwBKxKISSsqAZh4BfHDgD0imAHUQhANYQYyMMRhMkUABYwkYY0gCGAGxgA3YoQjkwMEAhgBBAAoBJBXhoGAATEABlEAFREwsUADV2DBgACnhkFCpcAiJSNwB+KmhxMBQAbl58fmFRcVYimBqAI2dK2QBKDAA+bCq4cAgQW2IAOlsQFFSAcgBVVzKBxFQKaa4lzI72nHComNN4pPTwTcqlFQSAJioAFRMDP384E2sy1jywawF4mFMDEAADlA3HYYADCIDnFAAJ5NQQiMQSHRQEAwATWAAe3igUGIAiBEFGRKCg1gu2isUOlxSx2WWT6uRIZHAjRK8UquXRWJxeIJjRabRwnR6DIIwWGYwmUzmCzQGRWaw2qC2QvOMDoJgYIBM6wOOmgSFs9hhAOIwHuwG0MGITncphAdBQZhY0xIYOsEFl5ncNQR9QgAEIdpEKQcyldaacgA) illustrates how changing the name of optional properties doesn't cause a type error.

Using the type derived above this does become a type error.

Reply:
Not blocking, only sharing because I got stuck with this recently! The v3 SDK seems to be [more customisable regarding retries](https://github.com/smithy-lang/smithy-typescript/blob/main/packages/util-retry/README.md).